torch/csrc/jit/script/parser.h

```diff
@@ -228,17 +228,17 @@ struct Parser {
       if (L.nextIf(']')) {
         return Gather::create(range, Expr(value), Expr(first));
       } else {
-        first = c(TK_OPTION, range, {first});
+        first = c(TK_NOTHING, range, {first});
       }
     } else {
-      first = c(TK_OPTION, range, {});
+      first = c(TK_NOTHING, range, {});
     }
     L.expect(':');
     // Now we *may* have an expression.
     if (L.cur().kind != ']') {
-      second = c(TK_OPTION, range, {parseExp()});
+      second = c(TK_NOTHING, range, {parseExp()});
```

Comment:
Huh? This was correct before. This is an option node with a single value _present_. It is certainly not nothing. TK_OPTION has a single subtree in the event a value is present, and no subtrees otherwise. Can you change this back?

Reply:
Good point. I changed it without reading into it, based on the comment in tree views. This means that the Maybe node was broken since the beginning. Will fix.